lib/vanagon/component.rb

```diff
@@ -92,6 +92,15 @@ class Component
     # preinstall_actions is a two-dimensional Array, describing scripts that
     # should be executed before a given component is installed.
     attr_accessor :preinstall_actions
+    # install_triggers is a one-dimensional Array of OpenStructs, describing scripts that
+    # should be executed when a package is installed or upgraded
+    attr_accessor :install_triggers
+    # interest_triggers is a one-dimensional Array of OpenStructs, describing scripts that
+    # should be executed when a package identifies an interest trigger
+    attr_accessor :interest_triggers
+    # activate_triggers is a one-dimentional Array of Strings, describing scripts that
+    # should be executed when a package identifies an activate trigger
+    attr_accessor :activate_triggers
```

Comment:
Why is this an Array of Strings while the others are Arrays of OpenStructs?

Reply:
Activate triggers are names used to track packages while interest triggers have names and scripts we want to run. This page explains it better than I do https://manpages.debian.org/jessie/dpkg-dev/deb-triggers.5.en.html

Reply:
ookay! 👍